sycl/tools/sycl-ls/sycl-ls.cpp

```diff
@@ -106,7 +106,7 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
-  auto Platforms = platform::get_platforms();
+  const auto &Platforms = platform::get_platforms();
```

Comment:
Could you please elaborate more on how the change ensures that lifetime is correct? What's wrong with current code?

Reply:
It makes the vector of platforms/devices returned by the get_platfroms/get_devices live until the reference to them is out of scope. Currently a copy of vector was made, and the original returned vector destructed.

Reply:
Since sycl classes are reference countable, lifetime is the same in both cases.

Reply:
OK, then this must be NFC, and I reflected this in the title.